### PR TITLE
Peak functions

### PR DESCRIPTION
**Pull request by jjgomezcadenas**
This PR addresses modifications in various functions needed to deal with the fact that one can have a number of dead PMTs (in real data). This is handled by defining a list of active PMTs which is passed around the functions that operate with PMTs. If the list is empty, the default is all PMTs active (this guarantees backward compatibility, e.g, preserves the signatures of functions). Some tests have been added to check the new functionality.

### CODE REVIEW

**Review by jmbenlloch — COMMENTED** [review]
This is a very important update since we are running with only 11 PMTs instead of 12. With this changes IC can process only active PMTs, fixing the errors we were getting when we had (inactive) PMTs with gain 0.

This PR includes some test for the new functionality and they work correctly. The only thing needed before approval is to squash the commits. @jjgomezcadenas can you do that?

**Comment by jjgomezcadenas** [comment]
This has been squashed (but somehow the conflict has reappeared)

**Review by jmbenlloch — APPROVED** [review]

**Comment by jmbenlloch** [comment]
I have solved the conflict and approved this PR.

**Pull request merged**
